### PR TITLE
feat: add a cap for transfer fee

### LINKED ISSUE #1995: Transfer fee improvements/reduction
## TODO:
- Only charge transfer fees if not offset by last epoch fees paid/generated by sender [for first N transfers?]?

We need to update the spec to detail that for each party cumulate the fees paid over an epoch, store the value if there is a transfer subtract the fee from the stored value. When this hits zero party gets charged. Need API support so front end can make it clear how much is transferred for free.

## Done:
- Cap transfer fees for larger transfers (multiple of the minimum amount) (in  feat: add a cap for transfer fee)

### PR DESCRIPTION
**Pull request by Jiajia-Cui**
closes  #1995

### CODE REVIEW

File: protocol/0057-TRAN-transfers.md [hunk 1]
```diff
@@ -156,6 +156,7 @@ Note: if there is no market with contribution to the reward metric - no transfer
 A fee is taken from all transfers, and paid out to validators in a similar manner to the existing [infrastructure fees](0061-REWP-pos_rewards.md). For recurring transfers, the fee is charged each time the transfer occurs.
 
 The fee is set by the `transfer.fee.factor` [network parameter](#network-parameters) that defines the proportion of each transfer taken as a fee.
+The fee is capped by the `transfer.fee.Maxfactor` network parameter to prevent people from paying a large transfer fee amount.
```
Comment (TomMcL) [inline protocol/0057-TRAN-transfers.md]: Should probably be something like `transfer.fee.maxQuantumAmount` as I expect we want it determined by some multiplier of a quantum?

File: protocol/0057-TRAN-transfers.md [hunk 2]
```diff
@@ -155,7 +155,7 @@ Note: if there is no market with contribution to the reward metric - no transfer
 
 A fee is taken from all transfers, and paid out to validators in a similar manner to the existing [infrastructure fees](0061-REWP-pos_rewards.md). For recurring transfers, the fee is charged each time the transfer occurs.
 
-The fee is set by the `transfer.fee.factor` [network parameter](#network-parameters) that defines the proportion of each transfer taken as a fee.
+The fee is determined by the `transfer.fee.factor` and is subject to a cap defined by the multiplier `transfer.fee.maxQuantumAmount` as specified in the network parameters, which governs the proportion of each transfer taken as a fee.
```
Comment (TomMcL) [inline protocol/0057-TRAN-transfers.md:158]:
```suggestion
The fee is determined by the `transfer.fee.factor` and is subject to a cap defined by the multiplier `transfer.fee.maxQuantumAmount` as specified in the network parameters, which governs the proportion of each transfer taken as a fee.

As such, the transfer fee value used will be: `min(transfer amount * transfer.fee.factor, transfer.fee.maxQuantumAmount * asset quantum)`
```

**Review by TomMcL — DISMISSED** [review]

File: protocol/0057-TRAN-transfers.md [hunk 3]
```diff
@@ -217,6 +219,7 @@ message CancelTransfer {
 | `spam.protection.maxUserTransfersPerEpoch` | String (integer) | strictly greater than `0`   | `"20"`         | The most transfers a use can initiate per epoch |
 | `transfer.minTransferQuantumMultiple`      | String (decimal) | greater than or equal to `0`| `"0.1"`        | This, when multiplied by `quantum` (which is specified per asset) determines the minimum transfer amount |
 | `transfer.fee.factor`                      | String (decimal) | in `[0.0,1.0]`              | `"0.001"`      | The proportion of the transfer charged as a fee  |
+| `transfer.fee.maxQuantumAmount`            | String (decimal) | in `[0.0,1.0]`              | `"0.001"`      | The multiplier of the transfer fee  |
```
Comment (TomMcL) [inline protocol/0057-TRAN-transfers.md]: Needs a wider range

**Review by TomMcL — APPROVED** [review]

**Comment by gordsport** [comment]
ACs to be added in this PR: https://github.com/vegaprotocol/specs/pull/2024

**Pull request merged**